### ISSUE TITLE
Parsing of I from Julia expressions

### ISSUE THREAD

**Issue by ChrisRackauckas** [symengine/symengine#1462]
`ex = SymEngine.Basic(:(2I^2 + 1))` outputs `-1` from SymEngine.jl. This is odd because `I` is not used for imaginary numbers in Julia since `im` is the imaginary. This breaks code which is derived from valid Julia expressions. I suggest switching the imaginary parsing to `im` for expressions from Julia.

**Comment by certik** [comment]
@ChrisRackauckas good point. I think the parser should have some options about this, and the Julia wrappers should use the options appropriate for Julia.

**Comment by isuruf** [comment]
How about a dictionary for constants? We could have `{"I": Symbol("I"), "im": Constant("I")} as a default for Julia.

**Comment by certik** [comment]
We already have that: https://github.com/symengine/symengine/blob/ed7479baf59a36636061acc35b676bcf9073a932/symengine/parser/parser.cpp#L218, it's just about making it configurable.

**Comment by isuruf** [comment]
Yeah, I meant an external dictionary that would be looked up first before looking at the internal one. This is similar to sympy's `local_dict` argument in `parse_expr`

**Comment by certik** [comment]
Right. It would look up this `local_dict` right before this line: https://github.com/symengine/symengine/blob/ed7479baf59a36636061acc35b676bcf9073a932/symengine/parser/parser.cpp#L224, and the `local_dict` can be an instance member variable of the Parser class, and by default if none is supplied, then looking things up in an empty dictionary might be very fast, that we might not incur any performance penalty in the default mode.
